ouroboros-network-framework/test/Test/Ouroboros/Network/Server2.hs

```diff
@@ -1730,9 +1730,12 @@ verifyRemoteTransition Transition {fromState, toState} =
 
       -- This might happen if starting any of the responders errored.
       (Nothing,           Nothing)           -> True
-      -- @RemoteWarmSt → RemoteWarmSt@ trnasition is observed if a hot or warm
-      -- protocol terminates (which triggers @RemoteEstSt -> RemoteWarmSt@)
+      -- @RemoteWarmSt → RemoteWarmSt@, @RemoteIdleSt → RemoteIdleSt@ and
+      -- @RemoteColdSt → RemoteColdSt@ transition are observed if a hot or
+      -- warm protocol terminates (which triggers @RemoteEstSt -> RemoteWarmSt@)
       (Just RemoteWarmSt, Just RemoteWarmSt) -> True
+      (Just RemoteIdleSt, Just RemoteIdleSt) -> True
+      (Just RemoteColdSt, Just RemoteColdSt) -> True
```

Comment:
Add a jira ticket for this bug. Describe there why these identity transitions are needed, link back to this thread and suggest that we need to update the specification.

Reply:
Basically this happens because in the code we are not making an effort to not log unspecified identity transitions. I do not see this as a bug, and only enabled these two transitions because you enabled `RemoteWarmSt -> RemoteWarmSt`. If you feel this shouldn't be allowed I can fix the code to ignore these transition cases.

Reply:
I added an issue for this, it's just a matter of specifying them explicitly in the documentation: #3402.